src/code/z_lifemeter.c

```diff
@@ -1,17 +1,446 @@
+    interfaceCtx->beatingHeartEnv[1] = (u8)(gFactor + 0x28) & 0xFF;
+    interfaceCtx->beatingHeartEnv[2] = (u8)(bFactor + 0x3C) & 0xFF;
+
+    sHeartsDDPrim[0][0] = 0xFF;
+    sHeartsDDPrim[0][1] = 0xFF;
+    sHeartsDDPrim[0][2] = 0xFF;
+
+    sHeartsDDEnv[0][0] = 0xC8;
+    sHeartsDDEnv[0][1] = 0x00;
+    sHeartsDDEnv[0][2] = 0x00;
+
+    sHeartsDDPrim[1][0] = sHeartsDDPrimColors[ddType][0];
+    sHeartsDDPrim[1][1] = sHeartsDDPrimColors[ddType][1];
+    sHeartsDDPrim[1][2] = sHeartsDDPrimColors[ddType][2];
+
+    sHeartsDDEnv[1][0] = sHeartsDDEnvColors[ddType][0];
+    sHeartsDDEnv[1][1] = sHeartsDDEnvColors[ddType][1];
+    sHeartsDDEnv[1][2] = sHeartsDDEnvColors[ddType][2];
+
+    rFactor = sHeartsDDPrimFactors[ddType][0] * ddFactor;
+    gFactor = sHeartsDDPrimFactors[ddType][1] * ddFactor;
+    bFactor = sHeartsDDPrimFactors[ddType][2] * ddFactor;
+
+    sBeatingHeartsDDPrim[0] = (u8)(rFactor + 0xFF) & 0xFF;
+    sBeatingHeartsDDPrim[1] = (u8)(gFactor + 0xFF) & 0xFF;
+    sBeatingHeartsDDPrim[2] = (u8)(bFactor + 0xFF) & 0xFF;
+
+    rFactor = sHeartsDDEnvFactors[ddType][0] * ddFactor;
+    gFactor = sHeartsDDEnvFactors[ddType][1] * ddFactor;
+    bFactor = sHeartsDDEnvFactors[ddType][2] * ddFactor;
+
+    sBeatingHeartsDDEnv[0] = (u8)(rFactor + 0xC8) & 0xFF;
+    sBeatingHeartsDDEnv[1] = (u8)(gFactor + 0x00) & 0xFF;
+    sBeatingHeartsDDEnv[2] = (u8)(bFactor + 0x00) & 0xFF;
+}
+
+s32 func_80100A80(GlobalContext* globalCtx) {
+    gSaveContext.health = (s16)globalCtx->interfaceCtx.unk_24E;
+    return 1;
+}
+
+s32 func_80100AA0(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    interfaceCtx->unk_250 = 0x140;
+    interfaceCtx->unk_24E = interfaceCtx->unk_24E + 0x10;
+    if (globalCtx->interfaceCtx.unk_24E >= gSaveContext.health) {
+        globalCtx->interfaceCtx.unk_24E = gSaveContext.health;
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80100AF0(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    s32 health;
+    if (interfaceCtx->unk_250 != 0) {
+        interfaceCtx->unk_250 -= 1;
+    } else {
+        interfaceCtx->unk_250 = 0x140;
+        interfaceCtx->unk_24E -= 0x10;
+        if (interfaceCtx->unk_24E <= 0) {
+            interfaceCtx->unk_24E = 0;
+            globalCtx->damagePlayer(globalCtx, -((health = gSaveContext.health) + 1));
+            return 1;
+        }
+    }
+    return 0;
+}
+
+extern TexturePtr D_02000000; // Empty heart texture
+extern TexturePtr D_02000400; // Full heart texture
+extern TexturePtr D_02000500; // Empty Double Defense Heart texture
+extern TexturePtr D_02000900; // Full Double Defense Heart texture
+extern TexturePtr heartQuarterTexture; 
+extern TexturePtr heartHalfTexture; 
+extern TexturePtr heartThreeQuarterTexture; 
+extern TexturePtr ddHeartQuarterTexture; 
+extern TexturePtr ddHeartHalfTexture; 
+extern TexturePtr ddHeartThreeQuarterTexture; 
```

Comment:
I recall from a previous review given to me that there was a reason to leave asses/undefined symbols as their addresses and to not give them names yet (leaving the names as comments is fine). I can't recall the reason, perhaps someone else could comment?